x/concentrated-liquidity/types/errors.go

```diff
@@ -851,3 +851,13 @@ type SwapNoProgressError struct {
 func (e SwapNoProgressError) Error() string {
 	return fmt.Sprintf("ran out of iterations during swap. Possibly entered an infinite loop. Pool id (%d), user provided coin (%s)", e.PoolId, e.UserProvidedCoin)
 }
+
+type SwapNoProgressWithConsumptionError struct {
```

Comment:
Would it be possible to come up with a test case that covers this new check?

I realize it might be difficult/not worth the time tradeoff but would be good to get an ack before merging that this is intentional

Reply:
I did not construct a direct unit test for this as it isn't worth the time. The difficulty stems from the swap APIs taking on sdk.Int amounts so it is more challenging to select numbers where the last swap step ends up with a tiny amount remaining such as `sdk.Dec`.

Similar cases are covered at the swap strategy level of abstraction where it is easier to set up due to working directly with `sdk.Dec`